### PR TITLE
[lldb][split-dwarf] implement GetSeparateDebugInfo for SymbolFileOnDemand

### PR DESCRIPTION
**Pull request by zhyty**
Easy change to get `image dump separate-debug-info` working when using `symbols.load-on-demand`. Also added a line of space in the default table output.

Updated the testing in response to comments.

Added tests
```
bin/lldb-dotest -p TestDumpDwo
```

It's easy to verify this manually by running 
```
lldb --one-line-before-file "settings set symbols.load-on-demand true" <some_target>
(lldb) image dump separate-debug-info
...
```

### CODE REVIEW

File: lldb/test/API/commands/target/dump-separate-debug-info/dwo/TestDumpDwo.py [hunk 1]
```diff
@@ -130,3 +130,29 @@ def test_dwos_not_loaded_table_output(self):
                 "0x[a-zA-Z0-9]{16}\s+E\s+.*foo\.dwo",
             ],
         )
+
+    @skipIfRemote
+    @skipIfDarwin
+    @skipIfWindows
```
Comment (bulbazord) [inline lldb/test/API/commands/target/dump-separate-debug-info/dwo/TestDumpDwo.py:136]: This test doesn't actually debug a process right? It looks like you're creating a target, modifying a setting, and then making sure that LLDB can find the dwo files correctly.

If that's the case, this test should totally work on Darwin and Windows because you can debug targets that use DWOs on Darwin and Windows.

File: lldb/source/Commands/CommandObjectTarget.cpp [hunk 2]
```diff
@@ -2691,6 +2691,7 @@ class CommandObjectTargetModulesDumpSeparateDebugInfoFiles
                     "Found unsupported debug info type '%s'.\n",
                     type.str().c_str());
               }
+              strm.EOL();
```
Comment (bulbazord) [inline lldb/source/Commands/CommandObjectTarget.cpp:2694]: This change looks unrelated? Why is this here?

File: lldb/test/API/commands/target/dump-separate-debug-info/dwo/TestDumpDwo.py (hunk 1 above)
Comment (clayborg) [reply lldb/test/API/commands/target/dump-separate-debug-info/dwo/TestDumpDwo.py:136]: The standard clang won't emit .dwo files when you build on a mac unless you specify the target triple. Then you also run into cases where someone might not have your architecture enabled in clang so it won't be able to build with the specified triple. I haven't looked at the Makefile for this, but I am guessing it is building with the clang we built, which we know supports the default triple, but we don't know if it will support a specified triple that allows .dwo files to be created. Any darwin clang producing darwin binaries won't make .dwos

Comment (bulbazord) [reply lldb/test/API/commands/target/dump-separate-debug-info/dwo/TestDumpDwo.py:136]: In that case, we should figure out if we can build the binaries for Linux when on a Darwin or Windows so that we can test this functionality on more than just Linux. I'm nervous about continuing to add functionality that can work on every host but only testing it on some hosts (in this case, dwo support). I work primarily on Darwin hosts and want to make sure I don't break this functionality on accident, ideally before committing a breaking change.

File: lldb/source/Commands/CommandObjectTarget.cpp (hunk 2 above)
Comment (zhyty) [reply lldb/source/Commands/CommandObjectTarget.cpp:2694]: The default `image dump separate-debug-info` table output is a bit cramped without the extra space. I mentioned this in the PR summary 
> Also added a line of space in the default table output.
But I can move this change to a separate PR, sorry!

File: lldb/test/API/commands/target/dump-separate-debug-info/dwo/TestDumpDwo.py (hunk 1 above)
Comment (zhyty) [reply lldb/test/API/commands/target/dump-separate-debug-info/dwo/TestDumpDwo.py:136]: I added a symbol-on-demand test to `TestDumpOso` for Darwin. I don't have a Windows machine, so I'm not sure how I would go about testing it there

**Review by clayborg — COMMENTED** [review]
Looks good to me.

FYI: here is how you might be able to try cross compiling by editing the Makefile and removing the `@skipIf*`. The compile command for cross compiling looks like:
```
clang++ -target x86_64-pc-linux-elf -g -gsplit-dwarf -c main.cpp
```
You will want to try adding a bogus `-target x86_64--carp` and make sure the test stops or can handle not compiling correctly. This will help you figure out what will happen if `-target x86_64-pc-linux-elf` doesn't work because ELF or linux support isn't compiled into clang. If you can't gracefully handle this case, then just leave the `@skip` decorators as is

**Comment by zhyty** [comment]
@bulbazord @clayborg I updated the DWO tests so that we try to compile for `x86_64-pc-linux-elf` on all platforms, and if we fail the tests are skipped. I've verified that the tests pass on my Linux machine and that they're skipped on my Mac, though it would be nice if I could make sure the build bots will pass.

**Review by clayborg — APPROVED** [review]
Looks good!
